src/textual/reactive.py

```diff
@@ -169,53 +159,41 @@ def _reset_object(cls, obj: object) -> None:
         getattr(obj, "__watchers", {}).clear()
         getattr(obj, "__computes", []).clear()
 
-    def __set_name__(self, owner: Type[MessageTarget], name: str) -> None:
-
+    def __set_name__(self, owner: Reactable, name: str) -> None:
         # Check for compute method
         if hasattr(owner, f"compute_{name}"):
             # Compute methods are stored in a list called `__computes`
-            try:
-                computes = getattr(owner, "__computes")
-            except AttributeError:
+            if not hasattr(owner, "__computes"):
                 computes = []
                 setattr(owner, "__computes", computes)
+            else:
+                computes = getattr(owner, "__computes")
             computes.append(name)
 
         # The name of the attribute
         self.name = name
         # The internal name where the attribute's value is stored
         self.internal_name = f"_reactive_{name}"
-        default = self._default
-        setattr(owner, f"_default_{name}", default)
 
     def __get__(self, obj: Reactable, obj_type: type[object]) -> ReactiveType:
         _rich_traceback_omit = True
 
         self._initialize_reactive(obj, self.name)
-
-        value: ReactiveType
-        compute_method = getattr(self, f"compute_{self.name}", None)
```

Comment:
so this is weird. notice `self` here, where I believe the original meaning was `obj`. But then it would have caused a ton of bugs, since compute would be triggers on every set no matter what.
Anyway -- this never really is `True` (at least in this version of the code), since `self` is the `Reactive` descr, and that has no `compute_foo` attrs. So just delete the whole thing, and avoid side-effect bugs.

Reply:
Something has definitely gone awry here. We did decide that reactives should call compute methods explicitly when you access the attribute. Which is what this is intended to do.

I think the logic with `obj` needs to be restored. We also need to look at tests which clearly weren't covering this.

Good catch.

Reply:
IF `compute_foo` should be triggered _every single time_ you access then yes, this should be `obj` and that's it.

BUT that means `run_compute` is redundant, and some other logic in `__set__` should be deleted as well.
It also means that there's no real well-defined behavior for WATCHING a computed a reactive.
Finally, it also means that the documentation is wrong. In the docs right now it says the opposite, and alludes to computed values being cached, and being all re-computed when a reactive value changes.

Anyway -- I am going to submit another PR with cached computed values (as the docs say now), and you can decide how you'd like to proceed.